context/app/static/js/components/workspaces/NewWorkspaceDialog/hooks.ts

```diff
@@ -42,16 +41,27 @@ function useUserTemplatesAPI<T>({ templatesURL, config = { fallbackData: {} } }:
 function useWorkspaceTemplates(tags: string[] = []) {
   const { userTemplatesEndpoint } = useAppContext();
 
-  const queryParams = tags.map((tag, i) => `${i === 0 ? '' : '&'}tags=${encodeURIComponent(tag)}`).join('');
-
-  const url = `${userTemplatesEndpoint}/templates/jupyter_lab/?${queryParams}`;
+  const url = `${userTemplatesEndpoint}/templates/jupyter_lab`;
   const result = useUserTemplatesAPI<TemplatesResponse>({ templatesURL: url });
 
   const templates = result?.data?.data ?? {};
 
+  // Manually update tags and title for R templates
+  const updatedTemplates = Object.fromEntries(
+    Object.entries(templates).map(([key, template]) => {
+      const isRTemplate = template.job_types?.includes(R_JOB_TYPE);
+
+      const updatedTitle = isRTemplate ? `${template.title} (${R_TEMPLATE_TAG})` : template.title;
+      const updatedTags = [...template.tags, ...(isRTemplate ? [R_TEMPLATE_TAG] : [])];
+
+      return [key, { ...template, tags: updatedTags, title: updatedTitle }];
+    }),
+  );
```

Comment:
Perhaps this post-processing could be handled as part of the `useUserTemplatesAPI` hook to keep a single source of truth in case that output is consumed anywhere else. It could also be a separate hook in the same file.

Reply:
I'm not sure if this is applicable now that we've moved away from adding the tags on the front end - let me know what you think, though!

Reply:
Nope, it should be all set! Handling this on the API side is the ultimate "single source of truth" approach 🙂